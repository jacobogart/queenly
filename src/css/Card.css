/* Testing Purpose */
.Card {
  /* border: yellow solid 5px; */
}
/* ==============  */
.Card {
  background-color: #202828;
  box-shadow: #2e5657 5px 5px 5px;
  top: 40px;
  left: 3%;
  width: 90%;
  position: absolute;
  height: 110%;
  padding: 25px;
  display: grid;
  grid-template-areas: "Main Sub" "Gall Gall";
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 450px 200px;
}

.favoriteBtn {
  position: absolute;
  top: 43px;
  right: 83px;
  font-size: 16px;
  background: none;
  border: none;
}

.toggle-close {
  position: fixed;
  background: none;
  width: 50px;
  font-size: 35px;
  right: 77px;
  border: none;
  top: 145px;
}

.toggle-close i {
  color: lightgrey;
}

.Card p {
  color: lightgrey;
}

.Main_Info {
  grid-area: Main;
}

.Sub_Info {
  grid-area: Sub;
}

iframe {
  border: 0;
  position: relative;
  right: -8px;
  top: 40px;
  /* border-radius: 20px; */
  background-color: #153233;
  padding: 10px;
  height: 358px;
  width: 570px;
}

.Card .Gallery {
  grid-area: Gall;
}
