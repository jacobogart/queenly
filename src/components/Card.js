import React, { Component } from "react";

// * Component imports
import Show_Main from "./Show_Main.js";
import Queen_Main from "./Queen_Main.js";
import Bar_Main from "./Bar_Main.js";
import Sub_Info from "./Sub_Info.js";
import Gallery from "./Gallery.js";

// * CSS imports
import "../css/Card.css";

export default class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      icons: {
        webIcon: <i className="fas fa-globe" />,
        phoneIcon: <i className="fas fa-phone-square" />,
        instaIcon: <i className="fab fa-instagram" />,
        facebookIcon: <i className="fab fa-facebook-square" />,
        twitterIcon: <i className="fab fa-twitter-square" />,
        close: <i className="fas fa-times-circle" /> 
      },
    };
  }
  
  render() {
    const cardData = this.props.cardData;
    const { id } = cardData;
    let mainInfo, cardType;

    if (id < 1000) {
      mainInfo = 
        <Queen_Main 
        cardData={cardData}
        icons={this.state.icons}
        />;
      cardType="Queen";
    } else if (id.toString().slice(2) > 0) {
      mainInfo = 
        <Show_Main 
        cardData={cardData} 
        icons={this.state.icons}
        />;
      cardType = "Show";
    } else {
      mainInfo = 
        <Bar_Main 
        cardData={cardData}
        icons={this.state.icons}
        />;
      cardType = "Bar";
    }

    return (
      <section className="Card">
          <button className="toggle-close" 
            onClick={this.props.displaySplashPage}>{this.state.icons.close}</button>
          {mainInfo}
          <Sub_Info 
            cardData={cardData}
            cardType={cardType}
          />
          <Gallery 
            cardData={cardData}
            cardType={cardType}
            selectSearchResult={this.props.selectSearchResult}
            bars={this.props.bars}
            queens={this.props.queens}
          />
      </section>
    );
  }
}
