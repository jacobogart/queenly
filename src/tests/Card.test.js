import React from "react";
import ReactDOM from "react-dom";
import Card from "../components/Card.js";
import { shallow } from "enzyme";

const mockSelectSearchResult = jest.fn();
const mockDisplaySplashPage = jest.fn();


const mockBars = [{ shows: [] }];
const mockQueens = [];
const mockCardData = {
  "bio":
      "❤️🧡💛💚💙💜💙💚💛🧡❤️ Double-D list Celebrity📍Chicago Drag Queen and Opinionated Scum ❤️🧡💛💚💙💜💙💚💛🧡❤️ IMHO EP. 4 💋⬇️",
  "facebook": "https://www.facebook.com/auntycherrychan",
  "id": 121,
  "imageURL":
      "https://scontent-ort2-1.cdninstagram.com/vp/f0b2195af69b918c1baa1ed18c56d446/5D4AF54C/t51.2885-19/s320x320/51218190_377069173112153_1261515169956102144_n.jpg?_nc_ht=scontent-ort2-1.cdninstagram.com",
  "instagram": "https://www.instagram.com/aunty_chan/?hl=en",
  "name": "Aunty Chan",
  "shows": ["Drag Matinee"],
  "twitter": "https://twitter.com/aunty_chan?lang=en"
  };

describe("Card", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card
        cardData={mockCardData}
        displaySplashPage={mockDisplaySplashPage}
        bars={mockBars}
        queens={mockQueens}
        selectSearchResult={mockSelectSearchResult}
      />
    );
  });
  it("should match snapshop", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have default state", () => {
    expect(wrapper.state()).toEqual({
      icons: {
        webIcon: <i className="fas fa-globe" />,
        phoneIcon: <i className="fas fa-phone-square" />,
        instaIcon: <i className="fab fa-instagram" />,
        facebookIcon: <i className="fab fa-facebook-square" />,
        twitterIcon: <i className="fab fa-twitter-square" />,
        close: <i className="fas fa-times-circle" />
      }
    });
  });
  it("should invoke displaySplashPage when X button is clicked", () => {
    wrapper.find(".toggle-close").simulate('click');
    expect(mockDisplaySplashPage).toBeCalled();
  });

});
