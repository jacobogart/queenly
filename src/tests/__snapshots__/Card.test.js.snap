// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshop 1`] = `
<section
  className="Card"
>
  <button
    className="toggle-close"
    onClick={[MockFunction]}
  >
    <i
      className="fas fa-times-circle"
    />
  </button>
  <Queen_Main
    cardData={
      Object {
        "bio": "❤️🧡💛💚💙💜💙💚💛🧡❤️ Double-D list Celebrity📍Chicago Drag Queen and Opinionated Scum ❤️🧡💛💚💙💜💙💚💛🧡❤️ IMHO EP. 4 💋⬇️",
        "facebook": "https://www.facebook.com/auntycherrychan",
        "id": 121,
        "imageURL": "https://scontent-ort2-1.cdninstagram.com/vp/f0b2195af69b918c1baa1ed18c56d446/5D4AF54C/t51.2885-19/s320x320/51218190_377069173112153_1261515169956102144_n.jpg?_nc_ht=scontent-ort2-1.cdninstagram.com",
        "instagram": "https://www.instagram.com/aunty_chan/?hl=en",
        "name": "Aunty Chan",
        "shows": Array [
          "Drag Matinee",
        ],
        "twitter": "https://twitter.com/aunty_chan?lang=en",
      }
    }
    icons={
      Object {
        "close": <i
          className="fas fa-times-circle"
        />,
        "facebookIcon": <i
          className="fab fa-facebook-square"
        />,
        "instaIcon": <i
          className="fab fa-instagram"
        />,
        "phoneIcon": <i
          className="fas fa-phone-square"
        />,
        "twitterIcon": <i
          className="fab fa-twitter-square"
        />,
        "webIcon": <i
          className="fas fa-globe"
        />,
      }
    }
  />
  <Sub_Info
    cardData={
      Object {
        "bio": "❤️🧡💛💚💙💜💙💚💛🧡❤️ Double-D list Celebrity📍Chicago Drag Queen and Opinionated Scum ❤️🧡💛💚💙💜💙💚💛🧡❤️ IMHO EP. 4 💋⬇️",
        "facebook": "https://www.facebook.com/auntycherrychan",
        "id": 121,
        "imageURL": "https://scontent-ort2-1.cdninstagram.com/vp/f0b2195af69b918c1baa1ed18c56d446/5D4AF54C/t51.2885-19/s320x320/51218190_377069173112153_1261515169956102144_n.jpg?_nc_ht=scontent-ort2-1.cdninstagram.com",
        "instagram": "https://www.instagram.com/aunty_chan/?hl=en",
        "name": "Aunty Chan",
        "shows": Array [
          "Drag Matinee",
        ],
        "twitter": "https://twitter.com/aunty_chan?lang=en",
      }
    }
    cardType="Queen"
  />
  <Gallery
    bars={
      Array [
        Object {
          "shows": Array [],
        },
      ]
    }
    cardData={
      Object {
        "bio": "❤️🧡💛💚💙💜💙💚💛🧡❤️ Double-D list Celebrity📍Chicago Drag Queen and Opinionated Scum ❤️🧡💛💚💙💜💙💚💛🧡❤️ IMHO EP. 4 💋⬇️",
        "facebook": "https://www.facebook.com/auntycherrychan",
        "id": 121,
        "imageURL": "https://scontent-ort2-1.cdninstagram.com/vp/f0b2195af69b918c1baa1ed18c56d446/5D4AF54C/t51.2885-19/s320x320/51218190_377069173112153_1261515169956102144_n.jpg?_nc_ht=scontent-ort2-1.cdninstagram.com",
        "instagram": "https://www.instagram.com/aunty_chan/?hl=en",
        "name": "Aunty Chan",
        "shows": Array [
          "Drag Matinee",
        ],
        "twitter": "https://twitter.com/aunty_chan?lang=en",
      }
    }
    cardType="Queen"
    queens={Array []}
    selectSearchResult={[MockFunction]}
  />
</section>
`;
